src/win/fs.c

```diff
@@ -142,6 +142,57 @@ void uv_fs_init(void) {
 }
 
 
+static unsigned int fs__decode_wtf8_char(const char** input) {
+  unsigned int code_point;
+  unsigned char b = **input, b2, b3, b4;
+  if (b >= 0x00 && b <= 0x7F) {
+    code_point = b;
+  } else if (b >= 0xC2 && b <= 0xDF) {
+    assert((*input)[1]);
```

Comment:
This input is user text, so it shouldn't be an assert here. Let's report this back as `STATUS_INVALID_PARAMETER` / `UV_ENOTDIR` if the value `b2` doesn't have the correct bits set.